src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java

```diff
@@ -197,7 +197,7 @@ public boolean canAttemptToEstablishConnection() {
       || containsMacro(SalesforceConstants.PROPERTY_CONNECT_TIMEOUT));
   }
 
-  private void validateConnection(OAuthInfo oAuthInfo) {
+  private void validateConnection(@Nullable OAuthInfo oAuthInfo) {
     if (!canAttemptToEstablishConnection()) {
```

Comment:
The `canAttemptToEstablishConnection` check here can be replaced with a null check, since it is already checked while attempting to get OAuthInfo.